### PR TITLE
Update .travis.yml for better performance

### PR DESCRIPTION
**Pull request by rsignell-usgs**
Added some tips from
http://krunoknego.com/2015/03/07/how-to-configure-travis-for-jekyll-github-page/

Will these help overcome the recurring test failures on travis?

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,10 +3,20 @@ env:
...
-
+        - NOKOGIRI_USE_SYSTEM_LIBRARIES=true # speeds up installation of html-proofer“
 before_script:
     - git config --global user.email $GIT_EMAIL
     - git config --global user.name $GIT_NAME
+    
+# branch whitelist
+branches:
+    only:
+        - master
```
Comment (ocefpaf) [inline .travis.yml]: Makes perfect sense!

File: .travis.yml [hunk 2]
```diff
@@ -3,10 +3,20 @@ env:
...
+        
+language: ruby
+
+rvm:
+    - 2.1
```
Comment (ocefpaf) [inline .travis.yml]: If this does what I think it does we do not need line 23.

**Comment by rsignell-usgs** [comment]
On this page http://jekyllrb.com/docs/continuous-integration/ it says:


Be sure to exclude vendor from your _config.yml

Travis bundles all gems in the vendor directory on its build servers, which Jekyll will mistakenly read and explode on.
```
exclude: [vendor]
```

**Comment by rsignell-usgs** [comment]
Adding 
```
travis_retry jeky'l build
```

**Comment by rsignell-usgs** [comment]
@mattben, we still have some issues here with travis seg faulting or timing out on the ```jekyll build``` step, but @ocefpaf is working on resolving the issue.

**Comment by mattben** [comment]
@rsignell-usgs i'm learning to use travis on my own project currently. I think its super cool you guys are trying to set travis up to build the site along with building all the doc book html pages. I wish I could be more helpful but i'm glad to see you all taking control of the project. :+1:

**Pull request merged**